third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp

```diff
@@ -715,28 +674,72 @@ class TritonIntelGPURewriteTensorPointerPass
   void runOnOperation() override {
     ModuleOp mod = getOperation();
 
-    auto usedByLoadOrStoreOp = [](Value val) {
-      return llvm::any_of(val.getUsers(), [](Operation *user) {
-        return isa<tt::LoadOp, tt::StoreOp>(user);
-      });
-    };
+    DenseSet<Operation *> tensorPointersToRemove;
+    mod.walk([&](Operation *op) {
+      if (isa<tt::MakeTensorPtrOp>(op)) {
```

Comment:
Looks like we only care about MakeTensorPtr?
```suggestion
    mod.walk([&](tt::MakeTensorPtrOp op) {
```